projects/hslayers/src/common/layman/access-rights/layman-access-rights.component.ts

```diff
@@ -1,27 +1,184 @@
+    return this.allUsers.find((u) => u.username == username);
+  }
+
+  /**
+   * Remove user rights by filtering username from access_rights
+   * @param type - Access rights type (access_rights.read or access_rights.write)
+   * @param username - User username provided
+   */
+  removeUserRights(type: string, username: string): void {
+    this.access_rights[type] = this.access_rights[type].filter(
+      (u: string) => u != username
+    );
+  }
+
+  /**
+   * Get a translation based on the value of the right
+   * @param right - Value of the access right
+   */
+  getRightTranslation(right: string): string {
+    if (right == 'read') {
+      return this.hsLanguageService.getTranslation(
+        'SAVECOMPOSITION.readAccessRights',
+        this.app
+      );
+    } else if (right == 'write') {
+      return this.hsLanguageService.getTranslation(
+        'SAVECOMPOSITION.writeAccessRights',
+        this.app
+      );
+    }
+  }
+
+  /**
+   * Change access granting options (everyone or per_user)
+   * @param option - Access granting option
+   */
+  changeGrantingOptions(option: 'per_user' | 'everyone'): void {
+    if (option == this.grantingOptions.PERUSER) {
+      this.getAllUsers();
+    } else {
+      this.allUsers = [];
+      this.access_rights['access_rights.read'] = 'EVERYONE';
+      this.access_rights['access_rights.write'] = 'private';
+    }
+    this.currentOption = option;
+  }
+
+  /**
+   * Get all registered users from Layman's endpoint service
+   */
+  async getAllUsers(): Promise<void> {
+    const endpoint: HsEndpoint = this.hsLaymanService.getLaymanEndpoint();
+    if (endpoint?.authenticated) {
+      const url = `${endpoint.url}/rest/users`;
+      try {
+        this.allUsers = await lastValueFrom(
+          this.$http.get<LaymanUser[]>(url, {withCredentials: true}).pipe(
+            map((res: any[]) => {
+              return res
+                .filter((user) => user.username != endpoint.user)
+                .map((user) => {
+                  return {
+                    username: user.username,
+                    screenName: user.screen_name,
+                    givenName: user.given_name,
+                    familyName: user.family_name,
+                    middleName: user.middle_name,
+                    name: user.name,
+                  };
+                });
+            })
+          )
+        );
+        this.access_rights['access_rights.read'] = [endpoint.user];
+        this.access_rights['access_rights.write'] = [endpoint.user];
+      } catch (e) {
+        this.hsLog.error(e);
+      }
+    }
+  }
+
+  /**
+   * Filter users by name
+   * This function is passed as filter pipe function
+   */
+  userFilter = (item): boolean => {
+    const r = new RegExp(this.userSearch, 'i');
+    return r.test(item.name);
+  };
+
+  /**
+   * Refresh user list, when searching for specific user
+   */
+  refreshList(): void {
+    this.allUsers = Array.from(this.allUsers);
```

Comment:
How does this work? Isnt allUsers already an array?

Reply:
To be honest I don't know, I took it from our code, without it it, doesn't update the list.